crates/bifrost/src/providers/replicated_loglet/sequencer/mod.rs

```diff
@@ -156,6 +158,26 @@ impl<T: TransportConnect> Sequencer<T> {
         self.record_permits.available_permits()
     }
 
+    pub fn ensure_enough_permits(&self, required: usize) {
+        let mut available = self.max_inflight_records_in_config.load(Ordering::Relaxed);
+        while available < required {
+            let delta = required - available;
+            match self.max_inflight_records_in_config.compare_exchange(
+                available,
+                required,
+                Ordering::Release,
+                Ordering::Relaxed,
```

Comment:
Perhaps you want the failure to be Acquire if your success's store path is Release?

Reply:
I actually wasn't sure if it's needed.I know it's definitely a stronger guarantee (with Release) but I thought relax loading it won't be a problem if the store failed and then it will try again.

But I will switch to `Acquire` just to be sure.